sound/hda/hdac_stream.c

```diff
@@ -333,6 +333,7 @@ void snd_hdac_stream_release(struct hdac_stream *azx_dev)
 	spin_lock_irq(&bus->reg_lock);
 	azx_dev->opened = 0;
 	azx_dev->running = 0;
+	azx_dev->assigned_key = 0;
```

Comment:
Confused. `snd_hdac_stream_assign()` is looking for closed streams with the same key:

```
		if (azx_dev->opened)
			continue;
		if (azx_dev->assigned_key == key) {
			res = azx_dev;
			break;
		}
```

After this it seems like it will never find such streams, because once a stream is released its `.assigned_key` is set to an invalid value?